hash/src/pedersen/mod.rs

```diff
@@ -264,11 +272,18 @@ where
     // generate partial sums
     let mut partial_point = NonZeroAffineVar::new(p0.x.clone(), p0.y.clone()).into_projective();
     let mut res = Vec::new();
+    let x_bits = x.to_le_bits();
     #[allow(clippy::needless_range_loop)]
     for i in 0..256 {
-        let suffix = x.rsh(i);
+        //let suffix = x.rsh(i);
```

Comment:
This is where a lot of memory and time are consumed, so I changed the way to perform bit operations.